util/codec/bytes.go

```diff
@@ -78,7 +81,8 @@ func decodeBytes(b []byte, escapeFirst byte, escape byte, term byte) ([]byte, []
 		return nil, nil, errors.Errorf("insufficient bytes to decode value")
 	}
 
-	var r []byte
+	// Allocate more space to avoid unnecessary slice growing
+	r := make([]byte, 0, len(b)+20)
```

Comment:
what I mean is not extra 20 bytes, but the b here may contain multi large slices, so if we decode the first slice, we don't need to alloc whole len(b).